### ISSUE TITLE
augmentation.mix: RandomMixUpV2/RandomCutMixV2 label rows as mixed when batch_prob leaves the image unmixed

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4775]
`RandomMixUpV2` and `RandomCutMixV2` gate the image with `params["batch_prob"]`, but their class-label handlers do not read `batch_prob`. In a normal call the sampled gate is batch-wide, so images and labels agree. When a caller passes a `params` dict whose `batch_prob` selects only some rows (for example to replay a draw on a subset, or when building `params` by hand), the unselected images come back unchanged while their labels still describe a mix with the drawn partner and weight.

## Repro

```python
import torch, kornia, kornia.augmentation as K
print(kornia.__version__, torch.__version__)
torch.manual_seed(0)
x = torch.rand(4, 3, 8, 8); y = torch.tensor([0, 1, 2, 3])
for cls in (K.RandomMixUpV2, K.RandomCutMixV2):
    aug = cls(p=1.0, data_keys=["input", "class"])
    aug(x, y)
    params = dict(aug._params)
    params["batch_prob"] = torch.tensor([True, False, True, False])
    out, lab = aug(x, y, params=params)
    print(cls.__name__, "rows 1, 3 unchanged:", torch.equal(out[1], x[1]), torch.equal(out[3], x[3]))
    print(lab)
```

## Observed (kornia 0.9.0rc1 @ 57cd68e6f, torch 2.14.0, CPU)

```
RandomMixUpV2 rows 1, 3 unchanged: True True
tensor([[0.0000, 0.0000, 0.1812],
        [1.0000, 2.0000, 0.9980],
        [2.0000, 3.0000, 0.2036],
        [3.0000, 1.0000, 0.9992]])
RandomCutMixV2 rows 1, 3 unchanged: True True
tensor([[[0.0000, 1.0000, 0.3906],
         [1.0000, 2.0000, 0.7656],
         [2.0000, 3.0000, 0.5625],
         [3.0000, 0.0000, 0.3906]]])
```

Rows 1 and 3 are the input images, but their labels say they were mixed with rows 2 and 1 (MixUp) or 2 and 0 (CutMix) at the drawn weight.

## Expected

The label of an unselected row describes an unmixed image: its own class as both entries with weight 0 (or whatever encoding the class documents for "no mix"), consistent with the image. Equivalently, `apply_transform_class` applies the same `batch_prob` gate as `apply_transform`.

## Why this is a bug

The image and box handlers of the same classes honour `batch_prob`, and `MixAugmentationBaseV2` documents `batch_prob` as the gate of the result. A training loop that computes the mix loss from these labels gets a wrong target for every unselected row whenever `params` does not carry a batch-wide gate. The current behaviour is documented as-is in the `MixAugmentationBaseV2` Convention block.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).
